torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -120,44 +123,161 @@ def can_auto_functionalize(op: OperatorBase) -> bool:
     return True
 
 
+@dataclass
+class ViewInfo:
+    base_index: int
+    size: Any = None
+    stride: Any = None
+    storage_offset: Any = None
+    is_alias_to_base: bool = False
+
+
+def serialize_views_meta(
+    arg_names, arg_types, input_kwargs, output_kwargs, arg_to_base_index
+):
+    def serialize_single_view(prefix, tensor, base_index):
+        if tensor is None:
+            output_kwargs[f"{prefix}_base_index"] = None
+        elif tensor._base is None:
+            # if the tensor is an alias of base (not view), for simplicity we do not serialize view meta.
+            output_kwargs[f"{prefix}_base_index"] = base_index
+            return
+        else:
+            output_kwargs[f"{prefix}_base_index"] = base_index
+            output_kwargs[f"{prefix}_size"] = tensor.size()
+            output_kwargs[f"{prefix}_stride"] = tensor.stride()
+            output_kwargs[f"{prefix}_storage_offset"] = tensor.storage_offset()
+
+    for arg_name, arg_type in zip(arg_names, arg_types):
+        arg = input_kwargs[arg_name]
+        if isinstance(arg_type, torch.ListType):
+            if arg is None:
+                output_kwargs[f"_{arg_name}_length"] = None
+
+            output_kwargs[f"_{arg_name}_length"] = len(arg)
+            for i, elem in enumerate(arg):
+                serialize_single_view(
+                    f"_{arg_name}_{i}", elem, arg_to_base_index[arg_name][i]
+                )
+
+        elif isinstance(arg_type, (torch.TensorType, torch.OptionalType)):
+            serialize_single_view(
+                f"_{arg_name}",
+                input_kwargs[arg_name],
+                arg_to_base_index.get(arg_name, None),
+            )
+        else:
+            raise RuntimeError(f"Unsupported type {arg_type}")
+
+
+def deserialize_views_meta(arg_names, arg_types, input_kwargs, all_bases, pop_args):
```

Comment:
Why do we need `pop_args`? Can we refactor it to not? Giving this function a side effect makes it harder to reason about.

Reply:
discussed offline